### PR TITLE
Image searchbox

### PR DESCRIPTION
**Pull request by rgalite**
Picture search allows users to search for a record from an uploaded image.

Under the hood, it uses Amazon Rekognition API to detect items and text from an image file. When the app performs a search, the detected items are used as filters and the detected text is used as query terms.

### CODE REVIEW

File: instantsearch-image-search/react-instantsearch/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+### CSS
+
+You will find the default css files in `src/components` folder. Each CSS file name matches the association component file name.
+
+## Usage
+
+```jsx
+import { InstantSearch } from 'react-instantsearch-dom'
+import TagsBox from 'instantsearch-tagsbox-react'
```
Comment (Haroenv) [inline instantsearch-image-search/react-instantsearch/README.md]: this import probably should be:
```suggestion
import ImageSearchBox from './components/ImageSearchBox'
```

File: instantsearch-image-search/react-instantsearch/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "client",
```
Comment (Haroenv) [inline instantsearch-image-search/react-instantsearch/package.json]:
```suggestion
  "name": "image-searchbox",
```

File: instantsearch-image-search/react-instantsearch/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "version": "1.0.0",
+  "private": true,
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "lint": "eslint .",
+    "lint:fix": "npm run lint -- --fix",
+    "deploy": "netlify deploy -d ./build --prod",
```
Comment (Haroenv) [inline instantsearch-image-search/react-instantsearch/package.json]: maybe some things for this need to be in .env?

File: instantsearch-image-search/react-instantsearch/.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+REACT_APP_API_URL=
```
Comment (Haroenv) [inline instantsearch-image-search/react-instantsearch/.env]: this should probably be `.env.example` and `.env` in `.gitignore` of the root

**Comment by rgalite** [comment]
@Haroenv can we merge this PR?

File: instantsearch-image-search/react-instantsearch/src/components/debug-zone.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React, { Component, Fragment } from 'react'
+import './debug-zone.css'
+
+class DebugZone extends Component {
+  state = { hidden: true }
+
+  handleToggleClick = () => {
+    this.setState({ hidden: !this.state.hidden })
```
Comment (francoischalifour) [inline instantsearch-image-search/react-instantsearch/src/components/debug-zone.js]:
```suggestion
    this.setState(state => ({ hidden: !state.hidden }))
```

Reference: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

**Comment by francoischalifour** [comment]
Note that although the `dist/` folder is now ignored, Git keeps track of it.

You can remove it with [this solution](https://stackoverflow.com/a/19095988/9940315).

**Comment by rgalite** [comment]
Alright, good to go.
As it's not a library but a full app, I removed the Usage section from the README.

**Pull request merged**